src/promptflow-tracing/promptflow/tracing/_start_trace.py

```diff
@@ -72,12 +72,30 @@ def _is_tracer_provider_set() -> bool:
     return isinstance(trace.get_tracer_provider(), TracerProvider)
 
 
+def _append_span_processors(current_trace_provider: TracerProvider, tracer_provider: TracerProvider):
+    from opentelemetry.sdk.trace import ConcurrentMultiSpanProcessor, SynchronousMultiSpanProcessor
+
+    if not isinstance(current_trace_provider, TracerProvider):
+        return
+    if not isinstance(
+        current_trace_provider._active_span_processor, (ConcurrentMultiSpanProcessor, SynchronousMultiSpanProcessor)
+    ):
+        return
+
+    # Append all span processors to the new tracer provider
+    for processor in current_trace_provider._active_span_processor._span_processors:
+        tracer_provider.add_span_processor(processor)
+
+
 def _force_set_tracer_provider(tracer_provider: TracerProvider) -> None:
     from opentelemetry.trace import _TRACER_PROVIDER_SET_ONCE
 
     with _TRACER_PROVIDER_SET_ONCE._lock:
         _TRACER_PROVIDER_SET_ONCE._done = False
 
+    #  Append all span processors to the new tracer provider to make sure all span processors are kept
+    current_trace_provider = trace.get_tracer_provider()
+    _append_span_processors(current_trace_provider, tracer_provider)
     trace.set_tracer_provider(tracer_provider)
```

Comment:
trace provider can't be overrided,  instead of setting a new trace provider, we should append all span processora to the existing trace provider if it's already set.